rust/ql/lib/codeql/rust/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -53,7 +53,7 @@ module Input implements InputSig<Location, RustDataFlow> {
 
   RustDataFlow::ArgumentPosition callbackSelfParameterPosition() { result.isClosureSelf() }
 
-  ReturnKind getStandardReturnValueKind() { result = TNormalReturnKind() }
+  ReturnKind getStandardReturnValueKind() { result = TNormalReturnKind() and Stage::ref() }
```

Comment:
What does this change do?

Reply:
I don't know ;-) I just ported all the relevant looking bits from this PR https://github.com/github/codeql/pull/17412 to to Rust.

Reply:
Ah, this is the old trick to force fewer stages. We have some [internal docs](https://github.com/github/codeql-team/blob/main/docs/best-practices/cached-stages-pattern.md) on it, but overall it's no longer needed after the CodeQL evaluator has been changed in regards to stages.

Reply:
Thanks, let's just get rid of it for now, unless @asgerf disagrees.

Reply:
No objections from me (assuming performance looks fine).